src/test/java/org/broadinstitute/hellbender/tools/walkers/markduplicates/AbstractMarkDuplicatesCommandLineProgramTest.java

```diff
@@ -235,28 +237,40 @@ public void testTwoMappedPairsAndMappedSecondaryFragment() {
         tester.addMappedPair(1, 1, 100, false, false, ELIGIBLE_BASE_QUALITY);
         tester.addMappedPair(1, 1, 100, true, true, DEFAULT_BASE_QUALITY); // duplicate!!!
         tester.addMappedFragment(1, 200, false, DEFAULT_BASE_QUALITY, true); // mapped non-primary fragment
+        if (this instanceof MarkDuplicatesSparkIntegrationTest) {
```

Comment:
I really don't like these instance of checks .  Maybe instead of the instanceof, we can add a method to` AbstractMarkDuplicatesTester.doNotMarkUnmappedMates()` that will be overriden by the the spark tester to provide the commandline argument.